### ISSUE TITLE
Storing secrets in user settings

### ISSUE THREAD

**Issue by g3n35i5** [microsoft/vscode#119015]
I would like to know if it is planned that you can save secrets in the user settings. I think of an implementation like it was done for example at [Home Assistant](https://www.home-assistant.io/docs/configuration/secrets/): Passwords or tokens can be stored in the separate file `secrets.yaml` and referenced in the configuration with `!secret secretVariableName`.

```yaml
http_password: "YOUR_PASSWORD"
```
```yaml
homeassistant:
  auth_providers:
   - type: legacy_api_password
     api_password: !secret http_password
```
I am aware that there is already similar feature request https://github.com/microsoft/vscode/issues/15909 (Extending settings files), but I think that my feature request would at least be a useful addition to this approach.

**Comment by IllusionMH** [comment]
Could you please specify what are expected use cases for storing secrets in VS Code settings?

I think that extensions should use Secrets API (see #112249) to store sensitive information.

**Comment by g3n35i5** [comment]
Dear @IllusionMH , thank you very much for your quick reply! I was not aware of the existence of the secrets API until now. It seems to be the better way to manage passwords. Unfortunately there are some (actually good) plugins that do not use the secrets API and therefore (still) store access data in plain text. I could still imagine my proposed functionality to support some kind of parameterizable configuration files (It does not necessarily have to exist only the command `secret`).

Basically, my feature request is about making the `settings.json` file available to a large group of people without hesitation, for example within a company. The individual settings, if necessary user names or passwords could then be stored in a file that is excluded with the gitignore.

**Comment by RMacfarlane** [comment]
@g3n35i5 Thanks for taking the time to write this up, but I think this request is out-of-scope for VS Code since we already have the secrets API. This API will write to the OS keychain/credential manager so is more secure than writing to a json file, so we would like to encourage extension authors to use it as much as possible.

For sharing settings between users in a company, I think this request is captured by https://github.com/microsoft/vscode/issues/27972. Please let me know if I've misunderstood.

**Comment by g3n35i5** [comment]
Dear @RMacfarlane , thank you for your reply! in fact, I have not seen number https://github.com/microsoft/vscode/issues/27972 yet, in fact that is a good idea. Apart from that, https://github.com/microsoft/vscode/issues/15909 seems to me to be a good alternative to https://github.com/microsoft/vscode/issues/27972 or complements this idea very well. Unfortunately both issues don't seem to be actively planned for development, are there any plans to change this?

**Comment by RMacfarlane** [comment]
Yeah, you're right, there's not a lot of activity on those. They aren't planned to be worked on in the immediate future, but we should have a new version of our [roadmap](https://github.com/microsoft/vscode/wiki/Roadmap) soon which shows our longer term plans. I think better support for configuring VS Code for organizations is part of that. Issues that make it into our backlog are those we think are valuable and plan to complete in the long run
